Florida_Analysis_Py_File.py

```diff
@@ -0,0 +1,534 @@
+            == "Drug poisonings (overdose) Suicide (X60-X64)"
+        )
+    )
+]
+
+florida_deaths.head(5)
+florida_deaths_copy = florida_deaths.copy()
+# calculating the drug related death rates in Florida per 100000 people
+
+florida_deaths_copy["Overdose_Per_100K"] = (
+    florida_deaths_copy["Deaths"] / florida_deaths_copy["Population"]
+) * 100_000
+
+florida_deaths_copy.head(3)
+# calculating and displaying year and county wise results for drug related deaths in Florida
+
+florida_result = (
+    florida_deaths_copy.groupby(["Year", "County"])["Overdose_Per_100K"]
+    .sum()
+    .reset_index()
+)
+florida_result.head(5)
+# calculating and displaying yearwise results for drug related deaths in Florida
+
+florida_results_yearwise = florida_deaths_copy.groupby("Year")[
+    "Overdose_Per_100K"
+].mean()
+florida_results_yearwise
+
+florida_result_copy = florida_result.copy()
+
+# create a scale for number of years before and after 2010 (target year)
+def scale_years(year):
+    if year == 2007:
+        return -3
+    if year == 2008:
+        return -2
+    if year == 2009:
+        return -1
+    if year == 2010:
+        return 0
+    if year == 2011:
+        return 1
+    if year == 2012:
+        return 2
+    if year == 2013:
+        return 3
+
+
+florida_result_copy["Year Relative To Policy"] = florida_result_copy["Year"].apply(
+    lambda x: scale_years(x)
+)
+
+# double check no nulls in "year relative to policy"
+
+assert florida_result_copy["Year Relative To Policy"].isnull().sum() == 0
+
+# doing this in case the float == int comparison causes issues
+
+# split into before 2010 and after 2010
+
+florida_mortality_before_policy_change = florida_result_copy[
+    florida_result_copy["Year"] < 2010
+]
+florida_mortality_after_policy_change = florida_result_copy[
+    florida_result_copy["Year"] > 2010
+]
+
+from sklearn.linear_model import LinearRegression
+
+regressor_before = LinearRegression()
+regressor_after = LinearRegression()
+
+
+x_before = np.array(
+    florida_mortality_before_policy_change["Year Relative To Policy"]
+).reshape(-1, 1)
+y_before = np.array(
+    florida_mortality_before_policy_change["Overdose_Per_100K"]
+).reshape(-1, 1)
+
+x_after = np.array(
+    florida_mortality_after_policy_change["Year Relative To Policy"]
+).reshape(-1, 1)
+y_after = np.array(florida_mortality_after_policy_change["Overdose_Per_100K"]).reshape(
+    -1, 1
+)
+
+
+regressor_before.fit(x_before, y_before)
+regressor_after.fit(x_after, y_after)
+
+
+y_pred_before = regressor_before.predict(x_before)
+y_pred_after = regressor_after.predict(x_after)
+
+import matplotlib.pyplot as plt
+
+plt.xlim(-3, 3)
+# plt.ylim(0, 500)
+
+plt.xlabel("Year Relative To Policy Change Year (2010)")
+plt.ylabel("Deaths due to Drug Overdose (Per 100K)")
+plt.title("Pre-Post Analysis Of Drug Overdose Deaths in Florida")
+
+plt.plot(x_before, y_pred_before, color="blue")
+plt.plot(x_after, y_pred_after, color="red")
+
+plt.grid()
```

Comment:
Hey Annapurna, you might want to include legends for the four lines on the plot. For example, simply add params as following: plt.plot(X_b4, y_pred_b4, color="b", label="Washington before")

Reply:
Thanks Zhonglin for your observation. I will add the legend accordingly.

Reply:
Thanks for your nice and quickly reply! Good job.